Exareme-Docker/src/exareme/exareme-master/src/main/java/madgik/exareme/master/gateway/async/handler/HttpAsyncMiningQueryHandler.java

```diff
@@ -283,313 +257,54 @@ private void handleInternal(HttpRequest request, HttpResponse response, HttpCont
-            while (entries.hasNext()) {
-                Map.Entry<String, String[]> entry = entries.next();
-                String IP = entry.getKey();
-                String[] datasets = entry.getValue();
-                //if dataset exist in that Exareme node
-                if (Arrays.asList(datasets).contains(data)) {
-                    //and Exareme node not already added to list nodesToBeChecked
-                    if (!nodesToBeChecked.contains(IP))
-                        nodesToBeChecked.add(IP);
-                    flag = true;
-                    continue;
-                }
-            }
-            //if flag=false then dataset(s) provided by user are not contained in ANY Exareme node
-            if (!flag) {
-                notFoundDatasets.add(data);
-            }
-        }
-        //if notFoundDatasets list is not empty, there are dataset(s) provided by user not contained in ANY Exareme node
-        if (notFoundDatasets.size() != 0) {
-            StringBuilder notFound = new StringBuilder();
-            for (String ds : notFoundDatasets)
-                notFound.append(ds).append(", ");
-            String notFoundSring = notFound.toString();
-            notFoundSring = notFoundSring.substring(0, notFoundSring.length() - 2);
-            //Show appropriate error message to user
-            throw new DatasetsException("Dataset(s) " + notFoundSring + " not found for pathology " +pathology + "!");
-        }
-        return nodesToBeChecked;
-    }
-
-    private String preALgoExecutionChecks(HttpRequest request) throws UnsupportedHttpVersionException {
-        log.debug("Validate method ...");
-        RequestLine requestLine = request.getRequestLine();
-        String uri = requestLine.getUri();
-        String algorithmName = uri.substring(uri.lastIndexOf('/') + 1);
-        String method = requestLine.getMethod().toUpperCase(Locale.ENGLISH);
-
-        if (!"POST".equals(method)) {
-            throw new UnsupportedHttpVersionException(method + "not supported.");
-        }
-        log.debug("Posting " + algorithmName + " ...\n");
-        return algorithmName;
-    }
-
-    private HashMap<String, String> getAlgoParameters(HttpRequest request) throws IOException {
-
-        log.debug("Parsing content ...");
-        HashMap<String, String> inputContent = new HashMap<>();
-        List<Map> parameters = new ArrayList();
-        String content;
-
-        if (request instanceof HttpEntityEnclosingRequest) {
-            log.debug("Streaming ...");
-            HttpEntity entity = ((HttpEntityEnclosingRequest) request).getEntity();
-            content = EntityUtils.toString(entity);
-            if (content != null && !content.isEmpty()) {
-                parameters = new Gson().fromJson(content, List.class);
-            }
+        String availableDatasets;
+        log.debug("Checking worker with IP "+IP);
+        if (checkIP.isReachable(5000)) {
+            log.info("Host is reachable");
```

Comment:
Before submitting logs should be refactored, info will be shown on prod, debug won't.